src/PSRule.Rules.Azure/rules/Azure.AKS.Rule.ps1

```diff
@@ -108,17 +108,24 @@ Rule 'Azure.AKS.AutoScaling' -Type 'Microsoft.ContainerService/managedClusters',
 }
 
 # Synopsis: AKS clusters using Azure CNI should use large subnets to reduce IP exhaustion issues.
-Rule 'Azure.AKS.CNISubnetSize' -Type 'Microsoft.ContainerService/managedClusters', 'Microsoft.ContainerService/managedClusters/agentPools' -If { IsExport } -Tag @{ release = 'GA'; ruleSet = '2021_09'; } {
+Rule 'Azure.AKS.CNISubnetSize' -Type 'Microsoft.ContainerService/managedClusters', 'Microsoft.ContainerService/managedClusters/agentPools' -If { IsExport } -With 'Azure.AKS.AzureCNI' -Tag @{ release = 'GA'; ruleSet = '2021_09'; } {
```

Comment:
@ArmaanMcleod Adding the selector will now only run the rule for types of `Microsoft.ContainerService/managedClusters` which I agree is correct. As you have said `Export-AzRuleTemplateData` doesn't list agent pools separately and that isn't intended because existing agent pools would always be part of a cluster also included in the export.

Happy with the suggestion, I think we can just clean up the Type filter. Which isn't needed now with the selector.

```suggestion
Rule 'Azure.AKS.CNISubnetSize' -If { IsExport } -With 'Azure.AKS.AzureCNI' -Tag @{ release = 'GA'; ruleSet = '2021_09'; } {
```